### PR TITLE
Extra exception handling to catch interrupted errors

### PR DESCRIPTION
**Pull request by Dinty1**

### CODE REVIEW

File: jda4/src/main/java/me/scarsz/jdaappender/ChannelLoggingHandler.java [hunk 1]
```diff
@@ -260,17 +261,23 @@ private Message updateMessage() throws IllegalStateException {
...
-            JDA.Status status = channel.getJDA().getStatus();
-            if (executor == null || executor.isShutdown() || status == JDA.Status.SHUTTING_DOWN || status == JDA.Status.SHUTDOWN) {
-                // ignored, no-op
-                return currentMessage;
-            } else {
-                throw new RuntimeException(e);
+        } catch (Exception e) {
+            Throwable ex = e;
+            while (ex != null) {
+                if (ex instanceof InterruptedIOException || ex.getCause() instanceof InterruptedException) return;
```
Comment (Vankka) [inline jda4/src/main/java/me/scarsz/jdaappender/ChannelLoggingHandler.java]:
```suggestion
                if (ex instanceof InterruptedIOException || ex instanceof InterruptedException) return;
```

File: jda4/src/main/java/me/scarsz/jdaappender/ChannelLoggingHandler.java [hunk 2]
```diff
@@ -260,17 +261,23 @@ private Message updateMessage() throws IllegalStateException {
...
+                ex = ex.getCause();
             }
-        }
+            if (e instanceof ExecutionException) {
+                throw new RuntimeException(e);
+                if (e instanceof InterruptedException) {
```
Comment (Vankka) [inline jda4/src/main/java/me/scarsz/jdaappender/ChannelLoggingHandler.java]: This isn't needed anymore

**Review by Vankka — CHANGES_REQUESTED** [review]
Has to be duplicated for `editMessage`, this also does not compile

**Comment by Dinty1** [comment]
Still doesn't compile on my end but that might be because my dev environment is messed up.

File: jda4/src/main/java/me/scarsz/jdaappender/ChannelLoggingHandler.java [hunk 3]
```diff
@@ -239,37 +240,25 @@ private Message updateMessage() throws IllegalStateException {
...
-                return currentMessage;
-            } else {
-                throw new RuntimeException(e);
+        } catch (Exception e) {
+            Throwable ex = e;
+            while (ex != null) {
+                if (ex instanceof InterruptedIOException || ex instanceof InterruptedException) break;
+                ex = ex.getCause();
             }
+            return currentMessage;
```
Comment (Vankka) [inline jda4/src/main/java/me/scarsz/jdaappender/ChannelLoggingHandler.java]: Please throw here as it did previously and only return when a interrupt is hit

File: jda4/src/main/java/me/scarsz/jdaappender/ChannelLoggingHandler.java [hunk 4]
```diff
@@ -239,37 +240,25 @@ private Message updateMessage() throws IllegalStateException {
...
-                    return currentMessage;
-                } else {
-                    throw new RuntimeException(e);
+            } catch (Exception e) {
+                Throwable ex = e;
+                while (ex != null) {
+                    if (ex instanceof InterruptedIOException || ex instanceof InterruptedException) break;
+                    ex = ex.getCause();
                 }
+                return currentMessage;
```
Comment (Vankka) [inline jda4/src/main/java/me/scarsz/jdaappender/ChannelLoggingHandler.java]: Please throw here as it did previously and only return when a interrupt is hit

**Review by Vankka — CHANGES_REQUESTED** [review]
See in-line comments.

The exception handling matches on both these places that could be extracted to a method but not required

JDA5 module should also match these changes

File: jda4/src/main/java/me/scarsz/jdaappender/ChannelLoggingHandler.java [hunk 5]
```diff
@@ -16,6 +16,7 @@
 import org.jetbrains.annotations.Nullable;
 
 import java.io.Flushable;
+import java.io.InterruptedIOException;
 import java.io.PrintStream;
```
Comment (granny) [inline jda4/src/main/java/me/scarsz/jdaappender/ChannelLoggingHandler.java]:
```suggestion
import java.io.Flushable;
import java.io.PrintStream;
```

File: jda4/src/main/java/me/scarsz/jdaappender/ChannelLoggingHandler.java [hunk 6]
```diff
@@ -239,37 +240,22 @@ private Message updateMessage() throws IllegalStateException {
...
-            JDA.Status status = channel.getJDA().getStatus();
-            if (executor == null || executor.isShutdown() || status == JDA.Status.SHUTTING_DOWN || status == JDA.Status.SHUTDOWN) {
-                // ignored, no-op
-                return currentMessage;
-            } else {
-                throw new RuntimeException(e);
-            }
+        } catch (Exception e) {
+            if (this.isInterruptedException(e)) return currentMessage;
+            else throw new RuntimeException(e);
```
Comment (granny) [inline jda4/src/main/java/me/scarsz/jdaappender/ChannelLoggingHandler.java]:
```suggestion
            throw new RuntimeException(e);
```

File: jda4/src/main/java/me/scarsz/jdaappender/ChannelLoggingHandler.java [hunk 7]
```diff
@@ -239,37 +240,22 @@ private Message updateMessage() throws IllegalStateException {
...
             try {
                 return currentMessage.editMessage(full).submit().get();
-            } catch (ExecutionException e) {
-                Throwable cause = e.getCause();
-                if (cause instanceof ErrorResponseException
-                        && ((ErrorResponseException) cause).getErrorResponse() == ErrorResponse.UNKNOWN_MESSAGE) {
+            } catch (Exception e) {
+                if (this.isInterruptedException(e)) return currentMessage;
+                if (e.getCause() instanceof ErrorResponseException
+                        && ((ErrorResponseException) e.getCause()).getErrorResponse() == ErrorResponse.UNKNOWN_MESSAGE) {
```
Comment (granny) [inline jda4/src/main/java/me/scarsz/jdaappender/ChannelLoggingHandler.java]:
```suggestion
                if (this.isInterruptedException(e)) return currentMessage;
                Throwable cause = e.getCause();
                if (cause instanceof ErrorResponseException
                        && ((ErrorResponseException) cause).getErrorResponse() == ErrorResponse.UNKNOWN_MESSAGE) {
```

**Review by granny — CHANGES_REQUESTED** [review]
some nitpicks

**Review by Vankka — APPROVED** [review]

**Pull request merged**